### PR TITLE
Split proxy username and password as secrets

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

File: config/http_config.go [hunk 1]
```diff
@@ -88,6 +92,12 @@ func (c *HTTPClientConfig) Validate() error {
 	if c.BasicAuth != nil && (string(c.BasicAuth.Password) != "" && c.BasicAuth.PasswordFile != "") {
 		return fmt.Errorf("at most one of basic_auth password & password_file must be configured")
 	}
+	if c.ProxyURL.String() == "" && (string(c.ProxyUsername) != "" && string(c.ProxyPassword) != "") {
+		return fmt.Errorf("if proxy_username is set, proxy_url must be set")
+	}
+	if string(c.ProxyPassword) != "" && string(c.ProxyUsername) == "" {
+		return fmt.Errorf("if proxy_password is set, proxy_username must be set")
```
Comment (brian-brazil) [inline config/http_config.go]: Is that what the spec says?

File: config/http_config.go [hunk 2]
```diff
@@ -72,6 +72,10 @@ type HTTPClientConfig struct {
 	BearerTokenFile string `yaml:"bearer_token_file,omitempty"`
 	// HTTP proxy server to use to connect to the targets.
 	ProxyURL URL `yaml:"proxy_url,omitempty"`
+	// Username to use to connect to the targets using proxy.
```
Comment (brian-brazil) [inline config/http_config.go]: Are there any other auth methods possible?

**Comment by simonpasquier** [comment]
How about crerating a `ProxyURL` struct embedding `URL` (eg `type ProxyURL struct {URL}`) and implement its `MarshalYAML()` method so it replaces the [`URL.User`](https://golang.org/pkg/net/url/#Userinfo) field with an obfuscated value if it isn't empty?

**Comment by brian-brazil** [comment]
That's not how we deal with secrets elsewhere, and is getting a tad magic.

**Pull request closed without merging**